src/v/raft/replicate_entries_stm.cc

```diff
@@ -89,13 +91,30 @@ replicate_entries_stm::send_append_entries_request(
     auto opts = rpc::client_opts(append_entries_timeout());
     opts.resource_units = ss::make_foreign<ss::lw_shared_ptr<units_t>>(_units);
 
-    auto f = _ptr->_client_protocol
-               .append_entries(n.id(), std::move(req), std::move(opts))
-               .then([this](result<append_entries_reply> reply) {
-                   return _ptr->validate_reply_target_node(
-                     "append_entries_replicate", std::move(reply));
-               });
-    _dispatch_sem.signal();
+    auto f = _ptr->_follower_queues.get_append_entries_unit(n).then_wrapped(
```

Comment:
I think there would be quite a lot of simplification if instead of passing around vnode and looking it up in different indexes, a single object tracked state for each vnode and then we didn't have to repeat all that add/remove/config-update logic in the follower queue--it could focus on queueing.

Reply:
In my first implementation I did exactly this i.e. added queue to `follower_stats` I found it a little confusing since I need to change API of follower stats to be futures but I can change this quickly.